### PR TITLE
feat: userId to be converted to string for Router

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

>To unmarshal the payload at the server end we need to send the `userId` as a string and not as a number. In some destinations, we are not doing this and directly mapping the `userId` in the response payload being sent to the server.
For processor destinations, we are handling this thing inside `versionedRouter` but we did not do that for the router which caused an error at the server end. This PR converts the userId to string in `versionedRouter` for router destination.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by lvrach** [comment]
Can you add a test case, for the situation you described?

**Comment by anantjain45823** [comment]
> Can you add a test case, for the situation you described?

@lvrach Added. Please check

**Review by ItsSudip — APPROVED** [review]

**Review by sanpj2292 — DISMISSED** [review]

**Review by sanpj2292 — DISMISSED** [review]

**Review by ItsSudip — DISMISSED** [review]

File: src/versionedRouter.js [hunk 1]
```diff
@@ -487,11 +483,23 @@ async function routerHandleDest(ctx) {
...
+          } else if (batchedRequest && typeof batchedRequest === 'object') {
+            // eslint-disable-next-line no-param-reassign
+            listOut.batchedRequest = enrichTransformedEvent(batchedRequest);
+          }
+
+          if (hasProcMetadataForRouter) {
+            // eslint-disable-next-line no-param-reassign
+            listOut.metadata = routerDestHandler.processMetadataForRouter(listOut);
+          }
+        });
```
Comment (utsabc) [inline src/versionedRouter.js:502]: This piece of code is not clearly understandable. Cannot clearly make-out what are we trying to do here?
can we reduce the complexity of this code in someway?

**Review by utsabc — CHANGES_REQUESTED** [review]

File: src/versionedRouter.js (hunk 1 above)
Comment (anantjain45823) [reply src/versionedRouter.js:502]: This code is basically dealing with the case when batched request comes in form of array (ex: Marketo) or in form of object (ex: Mp). At last for every request we are processing metadata if it needs to be.

Comment (utsabc) [reply src/versionedRouter.js:502]: ok

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**